manifests/params.pp

```diff
@@ -15,101 +15,67 @@
     # Puppet 6+ only needs to manage the puppet service:
     $service_names = ['puppet']
   }
+  if $::osfamily == 'windows' {
+    $local_puppet_dir = windows_native_path("${::puppet_agent_appdata}/Puppetlabs")
+    $local_packages_dir = windows_native_path("${local_puppet_dir}/packages")
 
-  # The `is_pe` fact currently works by echoing out the puppet version
-  # and greping for "puppet enterprise". With Puppet 4 and PE 2015.2, there
-  # is no longer a "PE Puppet", and so that fact will no longer work.
-  # Instead check both the `is_pe` fact as well as if a PE provided
-  # function is available
-  $_is_pe = (getvar('::is_pe') or is_function_available('pe_compiling_server_version'))
+    $confdir = $::puppet_confdir
 
-  # In Puppet Enterprise, agent packages are provided by the master
-  # with a default prefix of `/packages`.
-  if $::osfamily != 'windows' {
-    $_source = $_is_pe ? {
-      true    => "https://${::servername}:8140/packages",
-      default => undef,
-    }
-  } else {
-    $_source = undef
-  }
+    $puppetdirs = [regsubst($confdir,'\/etc\/','/code/')]
+    $path_separator = ';'
 
-  $package_name = 'puppet-agent'
-  $install_dir = undef
-  $install_options = []
+    $user  = 'S-1-5-32-544'
+    $group = 'S-1-5-32-544'
+  } else {
+    $local_puppet_dir = '/opt/puppetlabs'
+    $local_packages_dir = "${local_puppet_dir}/packages"
 
-  case $::osfamily {
-    'RedHat', 'Debian', 'Suse', 'Solaris', 'Darwin', 'AIX': {
-      $local_puppet_dir = '/opt/puppetlabs'
-      $local_packages_dir = "${local_puppet_dir}/packages"
+    $confdir = '/etc/puppetlabs/puppet'
 
-      $confdir = '/etc/puppetlabs/puppet'
+    # A list of dirs that need to be created. Mainly done this way because
+    # Windows requires more directories to exist for confdir.
+    $puppetdirs = ['/etc/puppetlabs', $confdir]
 
-      # A list of dirs that need to be created. Mainly done this way because
-      # Windows requires more directories to exist for confdir.
-      $puppetdirs = ['/etc/puppetlabs', $confdir]
+    $path_separator = ':'
 
-      $path_separator = ':'
+    $user  = 0
+    $group = 0
+  }
+  $ssldir = "${confdir}/ssl"
+  $config = "${confdir}/puppet.conf"
 
-      $user  = 0
-      $group = 0
+  # The `is_pe` fact currently works by echoing out the puppet version
+  # and greping for "puppet enterprise". With Puppet 4 and PE 2015.2, there
+  # is no longer a "PE Puppet", and so that fact will no longer work.
+  # Instead check both the `is_pe` fact as well as if a PE provided
+  # function is available
+  $_is_pe = (getvar('::is_pe') or is_function_available('pe_compiling_server_version'))
```

Comment:
This is checking for either, and the comment says checking for both?

Reply:
does `getvar('::is_pe')` work when that fact is missing and Puppet is in strict mode?

Reply:
No clue

Reply:
I think that's the point of `getvar`, but I honestly can't remember. The docs on it aren't super clear how it behaves in strict mode

Reply:
If we just moved it, it's fine